snuba/query/matcher.py

```diff
@@ -0,0 +1,218 @@
+from __future__ import annotations
```

Comment:
Also, my rationale here (besides there just being multiple types in this file): I suspect this will often be implemented as `from snuba.query import matchers` to avoid namespace collisions with `snuba.query.expressions`.